models/git/commit_status.go

```diff
@@ -240,6 +241,60 @@ func GetLatestCommitStatus(ctx context.Context, repoID int64, sha string, listOp
 	return statuses, count, db.GetEngine(ctx).In("id", ids).Find(&statuses)
 }
 
+// GetLatestCommitStatusForPairs returns all statuses with a unique context for a given list of repo-sha pairs
+func GetLatestCommitStatusForPairs(ctx context.Context, repoIDs []int64, shas []string, listOptions db.ListOptions) (map[int64][]*CommitStatus, error) {
```

Comment:
Isn't that effectively
```suggestion
func GetLatestCommitStatusForPairs(ctx context.Context, repoIDsToLatestCommitSHAs map[int64]string, listOptions db.ListOptions) (map[int64][]*CommitStatus, error) {
```
?